lib/response.js

```diff
@@ -139,8 +139,8 @@ function generatePayload (response) {
 
   // Prepare payload parsers
   res.json = function parseJsonPayload () {
-    if (res.headers['content-type'].indexOf('application/json') < 0) {
-      throw new Error('The content-type of the response is not application/json')
+    if (!/^application\/(?<tree>([\w-]*\.)*)(?<subtype>[\w-]*\+)?json(.*)$/.test(res.headers['content-type'])) {
```

Comment:
I am thinking, would it be better to remove the check entirely.

The function here is more or less the same as `fetch`. And it should not block the parsing based on `Content-Type`.

Reply:
I am happy to remove it entirely in case that's a better way to approach it. But I assume this is not supposed to get too close to fetch (i.e. adding the content type if it wasn't set)?

Reply:
I means `.json` is just doing `JSON.parse` which is a helper function only (same as `fetch`).

Reply:
Let see what the others think about it. 
@fastify/core 

I am also fine for the current implementation. @WoH And it is missing document update.
https://github.com/fastify/light-my-request/blob/f895e9a3045da411a30cfe82435b4c284054bb9a/README.md?plain=1#L184

Reply:
Thanks for the hint, let me know if my attempts at an explanation that doesn't sound too confusing worked 😅

Reply:
Looks kind of a potential performance bottleneck. Also why are there named groups if you just use .test() and no .match()? 

Also safe-regex shows that the regex is potentially unsafe.

Reply:
Groups to have a better idea what is going on. The current benchmarks ignore this, so posting them didn't seem helpful, I assume 'application/json' is the most common content type by far, maybe check that first and skip the regex?

Reply:
I would just drop the check to be honest.

Reply:
+1 to dropping the check

Reply:
Dropped the check, let me know it if I should change anything else.